src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -59,9 +59,23 @@ public static bool RequiresReflectionMethodBodyScannerForAccess (LinkContext con
 			return context.Annotations.FlowAnnotations.RequiresDataFlowAnalysis (fieldDefinition);
 		}
 
-		bool ShouldEnableReflectionPatternReporting (MethodDefinition method)
+		bool ShouldEnableReflectionPatternReporting ()
 		{
-			return !_context.Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (method);
+			// Check if the current scope method has RequiresUnreferencedCode on it
+			// since that attribute automatically suppresses all trim analysis warnings.
+			// Check both the immediate origin method as well as suppression context method
+			// since that will be different for compiler generated code.
+			IMemberDefinition suppressionContextMember = _scopeStack.CurrentScope.Origin.SuppressionContextMember;
+			if (suppressionContextMember is MethodDefinition suppressionContextMethod &&
+				_context.Annotations.HasLinkerAttribute<RequiresUnreferencedCodeAttribute> (suppressionContextMethod))
+				return false;
+
+			IMemberDefinition originMember = _scopeStack.CurrentScope.Origin.MemberDefinition;
+			if (suppressionContextMember != originMember && originMember is MethodDefinition originMethod &&
```

Comment:
Why do we need to check both? I would have thought that just checking SuppressionContextMember would be enough.

Reply:
```C#
[UnconditionalSuppressMessage("IL2026")]
void Test()
{
    LocalFunction ();
   
    [UnconditionalSuppressMessage("IL2071")]
    void LocalFunction() {
        // Do something which produces 2026 and also 2071
    }
}
```

In this case the `LocalFunction` is a compiler generated code (the actual name is something long and weird). But it does have a user provided attribute on it - so we need to look there. But at the same time, the attribute applied to the outer method still needs to apply to the body of `LocalFunction` as well (this is how Roslyn implements `UnconditionalSuppressMessage`)

Reply:
Thanks!